src/main/java/io/jenkins/plugins/artifact_manager_jclouds/BlobStoreProvider.java

```diff
@@ -51,49 +51,83 @@
         GET, PUT;
     }
 
-    /** A constant for the blob path prefix to use. */
+    public static final class Part implements Serializable {
+        private final int partNumber;
+        private final String eTag;
+
+        public Part(int partNumber, String eTag) {
+            this.partNumber = partNumber;
+            this.eTag = eTag;
+        }
+
+        public int getPartNumber() {
+            return partNumber;
+        }
+
+        public String getETag() {
+            return eTag;
+        }
+    }
+
+    public interface MultipartUploader extends AutoCloseable {
+        @NonNull
+        URL toExternalURL(int partNumber) throws IOException;
+
+        void complete(@NonNull List<Part> parts) throws IOException;
+    }
+
+    /**
+     * A constant for the blob path prefix to use.
+     */
     @NonNull
     public abstract String getPrefix();
 
-    /** A constant for the blob container name to use. */
+    /**
+     * A constant for the blob container name to use.
+     */
     @NonNull
     public abstract String getContainer();
 
-    /** A constant to define whether we should delete artifacts or leave them to be managed on the blob service side. */
+    /**
+     * A constant to define whether we should delete artifacts or leave them to be managed on the blob service side.
```

Comment:
Please minimize diffs when filing a PR—do not edit unrelated lines.

Reply:
Fixed